libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/SessionCredentialsService.java

```diff
@@ -88,6 +96,47 @@ public void persistCredentials(
         }
     }
 
+    public void deleteSessionCredentialsForSubjourneyType(
+            String ipvSessionId, CoiSubjourneyTypes coiSubjourneyTypes)
+            throws VerifiableCredentialException {
+        try {
+            List<SessionCredentialItem> sessionCredentialItems = dataStore.getItems(ipvSessionId);
+
+            List<SessionCredentialItem> vcsToDelete;
+
+            if (isAddressOnlyJourney(coiSubjourneyTypes)) {
+                vcsToDelete =
+                        sessionCredentialItems.stream()
+                                .filter(
+                                        item ->
+                                                item.getCriId().equals(ADDRESS_CRI)
+                                                        || item.getCriId()
+                                                                .equals(EXPERIAN_FRAUD_CRI))
+                                .collect(Collectors.toList());
```

Comment:
With the below you could also pull the `List` out into a constant.

```suggestion
                vcsToDelete =
                        sessionCredentialItems.stream()
                                .filter(item -> List.of(ADDRESS_CRI, EXPERIAN_FRAUD_CRI).contains(item.getCriId()))
                                .toList();
```